README.md

```diff
@@ -1,16 +1,39 @@
-## Threadx Demo
-- Using stmcubeide version: 1.15.1 and an STM32f446RE
-- RTOS code is in Threadx-Demo/Core/Src/app_threadx.c
-- Using 3 threads with normal priority
-- Thread0 
-	- loads a message into a queue
-	- When the message is loaded a thread 0 specific count increments by 1
-	- Thread0 sum is the sum of all the random numbers it sends
-	- Thread0 prints every global piece of data
-- Thread1 and thread2 
-	- Each take the message with the random number they recieved and print it over uart
-	- Thread1 sum is the sum of the numbers it recieves
-	- Shared count is a count of the total messages sent
-	- Shared sum is a sum of all the random numbers sent
-	- When the message is recieved the threads increment a gloabl variable by 1
-	- Thread1 and 2 print their resepective counts and messages recieved over uart
+# ThreadX Demo
+- Using STM32CubeIDE version 1.15.1 and an STM32F446RE
+- RTOS code is in `ThreadX-Demo/Core/Src/app_ThreadX.c`
+- Using 3 threads with normal priority:
+
+The point of this demo is to have an example of how we can create and use threads in ThreadX.
+
+## Thread 0:
+ 1. Generates a random number and, using tx_queue_send, adds it to a queue.
+ 2. Waits to be able to take a counting semaphore(When the code is first flashed the semaphore starts in the released state so thread 0 should always get it first).
+ 3. If the random number is successfully added to the queue a thread 0 specific count increments by 1 and the random number is added to a thread 0 sum.
+ 4. Releases a counting semaphore.
```

Comment:
It just takes a random semaphore? Also can probably just say a semaphore since its ability to count is a property of semaphores.

```suggestion
2. Waits to be able to take the counting semaphore(When the code is first flashed the semaphore starts in the released state so thread 0 should always get it first).
```
```suggestion
 4. Releases the counting semaphore.
```